transit_odp/timetables/transformers.py

```diff
@@ -161,9 +160,12 @@ def transform(self) -> TransformedData:
             ) = transform_service_pattern_to_service_links(  # noqa: E501
                 service_patterns, route_to_route_links, route_links
             )
-            service_patterns = service_patterns.drop_duplicates(
+            service_patterns = service_patterns.reset_index().drop_duplicates(
                 drop_duplicates_columns_sp
             )
+            service_patterns = service_patterns.set_index(
```

Comment:
Method `transform_stop_sequence` can be updated to ensure the geometries for routes are calculated only once. `route_hash` can be used to map geometry of unique row and then map these to the relevant service pattern row